blackjack/core.py

```diff
@@ -0,0 +1,278 @@
+    Queen = enum.auto()
+    King = enum.auto()
+
+
+class Card:
+    def __init__(self, rank: Rank, suit: Suit):
+        self.rank = rank
+        self.suit = suit
+
+    def __repr__(self):
+        return f"<Card({self.rank.name}, {self.suit.name})>"
+
+
+class GameState:
+    def __init__(self, deck: List[Card], player_hand: List[Card], dealer_hand: List[Card]):
+        self.deck = deck
+        self.player_hand = player_hand
+        self.dealer_hand = dealer_hand
+
+    def deal(self):
+        """Removes and returns first card of the deck."""
+        return self.deck.pop(0)
+
+
+def make_deck():
+    deck = []
+    for suit in Suit:
+        for rank in Rank:
+            card = Card(rank, suit)
+            deck.append(card)
+    return deck
+
+
+def house_hit(state: GameState):
+    while hand_total(state.dealer_hand) <= 17:
+        state.dealer_hand.append(state.deal())
+
+
+def hit_player(state: GameState):
+    state.player_hand.append(state.deal())
+
+
+def hit(hit_response, state: GameState):
+    if hit_response == Play.Hit:
+        hit_player(state)
+        return
+    if hit_response == Play.Stay:
+        house_hit(state)
+        return
+    raise ValueError(f"Invalid Play: {hit_response}")
+
+
+def hand_total(hand):
+    total = 0
+    ace_count = 0
+    for card in hand:
+        r = card.rank
+        total += card_point(card)
+        if r == Rank.Ace:
+            ace_count += 1
+        while ace_count > 0 and total > 21:
+            ace_count -= 1
+            total -= 10
+    return total
+
+
+def initial_deal(state):
+    state.player_hand.append(state.deal())
+    state.player_hand.append(state.deal())
+    state.dealer_hand.append(state.deal())
+    state.dealer_hand.append(state.deal())
+
+
+def render_hand(x):
+    ren_hand = ''
+    for card in x:
+        s = card.suit
+        r = card.rank
+        ren_card = ''
+        if r == Rank.Ace:
+            ren_card = "A"
+        if r == Rank.Two:
+            ren_card = '2'
+        if r == Rank.Three:
+            ren_card = '3'
+        if r == Rank.Four:
+            ren_card = '4'
+        if r == Rank.Five:
+            ren_card = '5'
+        if r == Rank.Six:
+            ren_card = '6'
+        if r == Rank.Seven:
+            ren_card = '7'
+        if r == Rank.Eight:
+            ren_card = '8'
+        if r == Rank.Nine:
+            ren_card = '9'
+        if r == Rank.Ten:
+            ren_card = '10'
+        if r == Rank.Jack:
+            ren_card = "J"
+        if r == Rank.Queen:
+            ren_card = "Q"
+        if r == Rank.King:
+            ren_card = "K"
+        if s == Suit.Spades:
+            ren_card += '♠ '
+            ren_hand += ren_card
+        if s == Suit.Hearts:
+            ren_card += '♥ '
+            ren_hand += ren_card
+        if s == Suit.Diamonds:
+            ren_card += '♦ '
+            ren_hand += ren_card
+        if s == Suit.Clubs:
+            ren_card += '♣ '
+            ren_hand += ren_card
```

Comment:
This does a lot of unnecessary `if` checks. Once one of them succeeds, you don't need to check the others.

Reply:
It might be easier to have a more functions; `render_suit`, `render_rank`, `render_card`; then `render_hand` will be much simpler.